adal/src/main/java/com/microsoft/aad/adal/BrokerAccountServiceHandler.java

```diff
@@ -324,6 +324,10 @@ private void bindToBrokerAccountService(final Context context, final Callback<Br
         if (brokerEvent != null) {
             brokerEvent.setBrokerAccountServiceBindingSucceed(serviceBound);
         }
+        if (!serviceBound) {
+            connection.unBindService(context);
+            callback.onError(new AuthenticationException(ADALError.BROKER_BIND_SERVICE_FAILED));
```

Comment:
Can we add an error log here.

Reply:
done. please check the updated PR. thanks